### PR TITLE
Add `trailing-newlines` example

### PR DESCRIPTION
**Pull request by harupy**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

Refs #5953

Closes #XXXX

### CODE REVIEW

**Comment by harupy** [comment]
Doc looks like this:

image

File: doc/data/messages/t/trailing-newlines/bad.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+print("apple")
+
```
Comment (harupy) [inline doc/data/messages/t/trailing-newlines/bad.py:2]:
```suggestion
print("apple")
# +3: [trailing-newlines]

```

worked. Is it correct?

Comment (Pierre-Sassoulas) [reply doc/data/messages/t/trailing-newlines/bad.py:2]: Maybe we could add a comment like for trailing-whitespace ? Something like this maybe:
```suggestion
print("apple")

# The file end with 2 lines that are empty # +2: [trailing-newlines]

```

Comment (Pierre-Sassoulas) [reply doc/data/messages/t/trailing-newlines/bad.py:2]:
```suggestion
print("apple")
:raw-html:`<br />`
```

Comment (harupy) [reply doc/data/messages/t/trailing-newlines/bad.py:2]: Is `:raw-html:`<br />` a special syntax to insert a newline?

Comment (Pierre-Sassoulas) [reply doc/data/messages/t/trailing-newlines/bad.py:2]: It's rst syntax to affect the html result that sphinx output. I did not test it but it should add a <br> which is a newline in HTML.

Comment (harupy) [reply doc/data/messages/t/trailing-newlines/bad.py:2]: @Pierre-Sassoulas 

```
`:raw-html:`<br />`
```

didn't work.

Comment (harupy) [reply doc/data/messages/t/trailing-newlines/bad.py:2]: I'm testing `literalinclude` directive:

https://www.sphinx-doc.org/ja/master/usage/restructuredtext/directives.html#directive-literalinclude

Comment (harupy) [reply doc/data/messages/t/trailing-newlines/bad.py:2]: image

Comment (harupy) [reply doc/data/messages/t/trailing-newlines/bad.py:2]: Required code change:

```diff
diff --git a/doc/exts/pylint_messages.py b/doc/exts/pylint_messages.py
index 0e5f13872..3ca0f467a 100644
--- a/doc/exts/pylint_messages.py
+++ b/doc/exts/pylint_messages.py
@@ -105,12 +105,10 @@ def _get_python_code_as_rst(code_path: Path) -> str:
     """
     if not code_path.exists():
         return ""
-    with open(code_path, encoding="utf-8") as f:
-        file_content = f.readlines()
     return f"""\
-.. code-block:: python
-
-{"".join("  " + i for i in file_content)}"""
+.. literalinclude:: /{code_path.relative_to(Path.cwd())}
+   :language: python
+"""
 
 
 def _create_placeholders(
```

Comment (harupy) [reply doc/data/messages/t/trailing-newlines/bad.py:2]: `_get_python_code_as_rst` looks like this:

```python
def _get_python_code_as_rst(code_path: Path) -> str:
    """Return the 'rst' representation of a python file or an empty string if the file
    does not exist.
    """
    if not code_path.exists():
        return ""
    return f"""\
.. literalinclude:: /{code_path.relative_to(Path.cwd())}
   :language: python
"""
```

Comment (Pierre-Sassoulas) [reply doc/data/messages/t/trailing-newlines/bad.py:2]: Neat !

File: doc/exts/pylint_messages.py [hunk 2]
```diff
@@ -105,12 +105,10 @@ def _get_python_code_as_rst(code_path: Path) -> str:
     """
     if not code_path.exists():
         return ""
-    with open(code_path, encoding="utf-8") as f:
-        file_content = f.readlines()
     return f"""\
-.. code-block:: python
-
-{"".join("  " + i for i in file_content)}"""
+.. literalinclude:: /{code_path.relative_to(Path.cwd())}
```
Comment (harupy) [inline doc/exts/pylint_messages.py:109]: `/{code_path.relative_to(Path.cwd())}` is equivalent to `/path/to/pylint/doc/{code_path.relative_to(Path.cwd())}`.

Sphinx doc says:

> The file name is usually relative to the current file’s path. However, if it is absolute (starting with /), **it is relative to the top source directory**.

File: doc/data/messages/t/trailing-newlines/bad.py (hunk 1 above)
Comment (harupy) [reply doc/data/messages/t/trailing-newlines/bad.py]: Generated rst file looks like this:

```rst
.. _trailing-newlines:

trailing-newlines / C0305
=========================

**Message emitted:**

Trailing newlines

**Description:**

*Used when there are trailing blank lines in a file.*

**Correct code:**

.. literalinclude:: /data/messages/t/trailing-newlines/good.py
   :language: python

**Problematic code:**

.. literalinclude:: /data/messages/t/trailing-newlines/bad.py
   :language: python



Created by the `format <https://github.com/PyCQA/pylint/blob/main/pylint/checkers/format.py>`__ checker.
```

**Review by Pierre-Sassoulas — APPROVED** [review]
Great solution to the problem we had 👍

**Pull request merged**